programs/quarter_board/main.c

```diff
@@ -0,0 +1,192 @@
+#include "ch.h"
+#include "hal.h"
+
+#include <icu_lld.h>
+#include "chprintf.h"
+#include <inttypes.h>
+
+
+#include "communication.h"
+
+#define BIG_BOOLSHIT 3000
+#define SMALL_BOOLSHIT 400
+#define NOT_SEEING_VALUE 3000
+#define QUANTITY_OF_SENSORS 2
+
+
+int32_t send_values[QUANTITY_OF_SENSORS];
+int32_t sensors_values[QUANTITY_OF_SENSORS];
+bool values_available[QUANTITY_OF_SENSORS];
+
+thread_t * commanderp;
+thread_t * ir_measurementp;
+
+/*
+ * ICU callbacks
+ */
+
+static void read_pulse(ICUDriver *icup) {
+    uint8_t i;
+    int32_t value;
+
+    if(icup == &ICUD1) {
+        i = 0;
+    } else {
+        i = 1;
+    }
+    
+    value = icuGetWidthX(icup);
+
+    if(value < BIG_BOOLSHIT && value > SMALL_BOOLSHIT) {    
+        sensors_values[i] = value;
+        values_available[i] = true;
+    } else {
+        values_available[i] = false;
+    }
+}
+
+THD_WORKING_AREA(waThread1, 64);
+THD_FUNCTION(Thread1, arg) {
+    (void)arg;
+
+    uint8_t i;
+
+    while (1) {
+        if(i_am_middle_one) {
+            values_available[0] = false;
+        }
+
+        for(i = 0; i < QUANTITY_OF_SENSORS; i++) {
+            if(values_available[i]) {
+                send_values[i] = sensors_values[i];
+                values_available[i] = false;
+            } else {
+                send_values[i] = NOT_SEEING_VALUE;
+            }
+        }
+        chThdSleepMilliseconds(100);
+    }
+    chRegSetThreadName("ir_measurement");
+    chThdSleepMilliseconds(100);
+}
+void execute_master_command(uint16_t command_id, uint8_t data_lenght, uint8_t *buff) {
+    switch(command_id) {
+        case 0xAA:  if(buff[0] == 1) led(GREEN, 1);
+                    else led(GREEN, 0);
+
+                    if(buff[1] == 1) led(RED, 1);
+                    else led(RED, 0);
+
+                    break;

```

Comment:
You are missing default clause, please add it (even if it would only be empty clause containing only `break`).